packages/web/src/lib/myalgowallet-mode.ts

```diff
@@ -0,0 +1,142 @@
+				}
+			})
+			.catch((err) => {
+				console.log(err);
+			});
+	}
+
+	// https://connect.myalgo.com/docs/interactive-examples/Connect
+	async connectToMyAlgo(): Promise<void> {
+		try {
+			this.accounts = await this.connector.connect({
+				shouldSelectOneAccount: true,
+				openManager: true,
+			});
+			this.addresses = this.accounts.map((account) => account.address);
+		} catch (err) {
+			throw new Error("Error while connecting to my algo wallet");
+		}
+	}
+
+	/**
+	 * https://connect.myalgo.com/docs/interactive-examples/PaymentTransaction
+	 * Sign a single transaction from a my algo wallet session
+	 * @param txn { SDK transaction object, shouldSign, signers, msig } object
+	 * @returns raw signed txn
+	 */
+	async signTransaction(txn: algosdk.Transaction): Promise<SignedTx> {
+		return await this.connector.signTransaction(txn.toByte());
+	}
+
+	/**
+	 * https://connect.myalgo.com/docs/interactive-examples/GroupedTransaction
+	 * Sign a group of transaction(s) from a my algo wallet session
+	 * @param txns { SDK transaction object, shouldSign, signers, msig } object
+	 * @returns array of raw signed txns | null. null representes that the txn in array is NOT signed
+	 * by wallet user (i.e signable by someone else).
+	 */
+	async signTransactionGroup(txns: TransactionInGroup[]): Promise<SignedTx[]> {
+		const txnsGroup = txns.map((v) => v.txn);
+		const groupID = algosdk.computeGroupID(txnsGroup);
+		for (let i = 0; i < txns.length; i++) txnsGroup[i].group = groupID;
+		return await this.connector.signTransaction(txnsGroup.map((txn) => txn.toByte()));
+	}
+
+	/**
+	 * Send signed transaction to network and wait for confirmation
+	 * @param rawTxns Signed Transaction(s)
+	 */
+	private async sendAndWait(
+		rawTxns: Uint8Array | Uint8Array[]
+	): Promise<algosdk.modelsv2.PendingTransactionResponse> {
+		const txInfo = await this.algodClient.sendRawTransaction(rawTxns).do();
+		return await this.waitForConfirmation(txInfo.txId);
+	}
+
+	// Function used to wait for a tx confirmation
+	private async waitForConfirmation(
+		txId: string
+	): Promise<algosdk.modelsv2.PendingTransactionResponse> {
+		/* eslint no-constant-condition: ["error", { "checkLoops": false }] */
+		const response = await this.algodClient.status().do();
+		let lastround = response[LAST_ROUND];
+		while (true) {
+			const pendingInfo = await this.algodClient.pendingTransactionInformation(txId).do();
+			if (pendingInfo["pool-error"]) {
+				throw new Error(`Transaction Pool Error: ${pendingInfo["pool-error"] as string}`);
+			}
+			if (pendingInfo[CONFIRMED_ROUND] !== null && pendingInfo[CONFIRMED_ROUND] > 0) {
+				return pendingInfo as algosdk.modelsv2.PendingTransactionResponse;
+			}
+			lastround++;
+			await this.algodClient.statusAfterBlock(lastround).do();
+		}
+	}
+
+	/**
+	 * Execute single transaction or group of transactions (atomic transaction)
+	 * @param execParams transaction parameters or atomic transaction parameters
+	 */
+	async executeTransaction(
```

Comment:
I have added a followup task (to be done after this pr will be merged): https://www.pivotaltracker.com/story/show/181626750